cli/configuration/index.md

```diff
@@ -0,0 +1,302 @@
+- `branches` - array of used branches
+  - `id` - ID of the branch
+  - `path` - name of the directory containing the branch configuration (e.g. `main`)
+- `configurations` - array of component configurations
+  - `branchId` - ID of the branch the configuration belongs to
+  - `componentId` - ID of the component (e.g. `keboola.ex-aws-s3`)
+  - `id` - ID of the configuration
+  - `path` - path to the configuration in the local directory (e.g. `extractor/keboola.ex-aws-s3/7241111/my-aws-s3-data-source`)
+  - `rows` - array of configuration rows (if the component supports rows)
+    - `id` - ID of the row
+    - `path` - path to the row from the configuration directory (e.g. `rows/7241227-cities`)
+
+
+## Directory Structure
+
+```
+├─ .gitignore
+├─ .env.local                           - contains API token
+├─ .env.dist                            - template for .env.local
+├─ .keboola                             - metadata
+   └─ manifest.json
+└─ [branch-id]-[branch-name]            - e.g. 641226045-test
+   ├─ meta.json                         - contains branch metadata
+   ├─ description.md                    - contains branch description
+   └─ [component-type]                  - e.g. extractor
+      └─ [component-id]                 - e.g. keboola.ex-db-oracle
+         └─ [config-id]-[config-name]   - e.g. 641226048-oauth-test
+            ├─ meta.json                - contains config metadata
+            ├─ config.json              - contains configuration
+            ├─ description.md           - contains config description
+            └─ rows
+               └─ [row-id]-[row-name]   - e.g. 28790-prod-fact-table
+                  ├─ meta.json          - contains row metadata
+                  ├─ config.json        - contains configuration
+                  └─ description.md     - contains description
+```
+
+## Branches
+
+The tool works with [dev branches](/components/branches/) by default. You can choose the branches from the project 
+you want to work with locally in [init](/cli/commands/init/) command. You can ignore dev branches concept and work with 
+the main branch only of course but note that all its configurations will be stored in the directory `main`.
+
+The directory of the main branch is called simply `main` and does not contain the branch id so that it is easily 
+recognizable from the other branches.
+
+The directory contains `description.md` where you can write description formatted in [Markdown](https://www.markdownguide.org/) 
+and `meta.json` containing the name of the branch and flag if it is default or not.
+
+Example of `meta.json`:
+```json
+{
+  "name": "Main",
+  "isDefault": true
+}
+```
+
+Then there are directories thematically grouping components: `extractor`, `other`, `transformation`, `writer`.
+
+Example of a branch folder with components configurations:
+
+{: .image-popup}
+![Screenshot -- A configuration directory example](/cli/configuration/directory-example.jpg)
```

Comment:
IDs budeme potrebovat vyhodit aj zo screenshotov, ... ak by sa to nestihalo, tak sa to spravi neskor.